backend/src/test/java/org/example/backend/controller/UserControllerTest.java

```diff
@@ -151,4 +153,33 @@ void updateFavoriteProjectsOfUserTest_RemoveProject() throws Exception {
 
         verify(userService, times(1)).removeFavoriteProject(any(User.class), anyString());
     }
+
+    @Test
+    @WithMockUser
+    void getAllUsersTest() throws Exception {
+        List<User> users = new ArrayList<>();
+
+        User user1 = new User();
+        user1.setId("1");
+        user1.setName("Test User 1");
+        users.add(user1);
+
+        User user2 = new User();
+        user2.setId("2");
+        user2.setName("Test User 2");
+        users.add(user2);
+
+        when(userService.getAllUsers()).thenReturn(users);
```

Comment:
Is it realy an integration test, if you mock the action outside the controller?

Reply:
You're right. I mocked the userService etc, to avoid authentication problems. But I now just mocked the user to bypass the authentication during my tests.